### PR TITLE
[onert] Revisit GenModelTrain

### PR DESCRIPTION
**Pull request by ragmani**
This commit revisits GenModelTrain in nnfw_api_tests.
   - Change TrainCaseData to mean one dataset
   - Allow TrainCaseData to add data by step or dataset
   - Apply changed TrainCaseData to Conv2D tests

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

---

Draft PR #12985

### CODE REVIEW

File: tests/nnfw_api/lib/GenModelTrain.h [hunk 1]
```diff
@@ -265,69 +368,73 @@ class GenModelTrain : public ::testing::Test
...
-        // Train
+        // Prepare expected losses
         const auto &ref_losses = train_case.losses;
-        std::vector<float> actual_losses(num_expecteds, 0.f);
+        std::vector<float> actual_losses;
+
         for (uint32_t epoch = 0; epoch < num_epoch; ++epoch)
         {
-          NNFW_ENSURE_SUCCESS(nnfw_train(_so.session, true));
+          actual_losses.resize(num_expecteds, 0.f);
```
Comment (jyoungyun) [inline tests/nnfw_api/lib/GenModelTrain.h:384]: There is no difference between the current size and given count, it does not initialize the value as `0.f`. It still contains the previous loss value, so now, the results between epochs are added together and printed out.

Here is the TensorFlow result with the same parameter:
```
Epoch 1/2 - loss: 403.3333
Epoch 2/2 - loss: 394.7772
```

Your loss `798.110534f` is the sum of the above two results.

File: tests/nnfw_api/lib/GenModelTrain.h [hunk 2]
```diff
@@ -20,22 +20,68 @@
...
   /**
-   * @brief A vector of losses buffers
+   * @brief A dataset of inputs/outputs
+   */
+  DataSet dataset;
+
+  /**
+   * @brief A vector of loss buffer
+   */
+  std::vector<float> losses;
```
[the remark this answers is not in the document]
Comment (jyoungyun) [reply tests/nnfw_api/lib/GenModelTrain.h:53]: (added)
Your loss value is the sum of the results between epochs. It seems that we should apply the above suggestion in this PR.

[the remark this answers is not in the document]
Comment (ragmani) [reply tests/nnfw_api/lib/GenModelTrain.h:53]: I agree with you :)

[the remark this answers is not in the document]
Comment (ragmani) [reply tests/nnfw_api/lib/GenModelTrain.h:53]: > Your loss value is the sum of the results between epochs. It seems that we should apply the above suggestion in this PR.

Thanks for you information. I apply it as your suggestion.

**Review by jyoungyun — APPROVED** [review]
LGTM

**Review by hseok-oh — APPROVED** [review]
